bot/seasons/easter/egg_hunt/cog.py

```diff
@@ -0,0 +1,511 @@
+        )
+
+        log.debug("Committing Super Egg scores.")
+        db.commit()
+        db.close()
+
+        embed.set_footer(text=f"Finished with {count} total reacts.")
+        with contextlib.suppress(discord.HTTPException):
+            await self.message.edit(embed=embed)
+
+    async def start_timeout(self, seconds=None):
+        """Starts the super egg session."""
+
+        if not seconds:
+            return
+        count = 4
+        for _ in range(count):
+            await asyncio.sleep(1)
+            embed = self.message.embeds[0]
+            embed.set_footer(text=f"Finishing in {count} minutes.")
+            try:
+                await self.message.edit(embed=embed)
+            except discord.HTTPException:
+                break
+            count -= 1
+        bot.remove_listener(self.collect_reacts, name="on_reaction_add")
+        await self.finalise_score()
+
+
+class EggHunt(commands.Cog):
+    """Easter Egg Hunt Event"""
+
+    def __init__(self):
+        self.event_channel = GUILD.get_channel(Channels.seasonalbot_chat)
+        self.task = asyncio.create_task(self.super_egg())
+        self.task.add_done_callback(self.task_cleanup)
+
+    @staticmethod
+    def task_cleanup(task):
+        """Returns a task result. Used as a done callback to show raised exceptions."""
+
+        task.result()
+
+    @staticmethod
+    def current_timestamp() -> int:
+        """Returns a timestamp of the current UTC time."""
+
+        return int(datetime.utcnow().replace(tzinfo=timezone.utc).timestamp())
+
+    async def super_egg(self):
+        """Manages the timing of super egg drops."""
+
+        while True:
+            now = self.current_timestamp()
+
+            if now > EggHuntSettings.end_time:
+                log.debug(f"Hunt ended. Ending task.")
+                break
+
+            if now < EggHuntSettings.start_time:
+                remaining = EggHuntSettings.start_time - now
+                log.debug(f"Hunt not started yet. Sleeping for {remaining}.")
+                await asyncio.sleep(remaining)
+
+            log.debug(f"Hunt started.")
+            current_window = EggHuntSettings.start_time
+            next_window = 0
+            for window in EggHuntSettings.windows:
+                window = int(window)
+                if window < now:
+                    current_window = window
+                    continue
+                if not next_window:
+                    next_window = window
+                else:
+                    break
+
+            log.debug(f"Current Window: {current_window}. Next Window {next_window}")
+
+            db = sqlite3.connect(DB_PATH)
+            c = db.cursor()
+            c.execute(f"SELECT COUNT(*) FROM super_eggs WHERE window={current_window}")
+            count = c.fetchone()[0]
+            db.close()
+
+            if not count:
+                log.debug(f"test")
```

Comment:
Why is this an f-string?

Reply:
Really missed this one, it should be removed entirely.